File: example/src/main/kotlin/com/xebia/functional/auto/ASCIIArt.kt
package com.xebia.functional.auto

import kotlinx.serialization.Serializable

@Serializable
data class ASCIIArt(val art: String)

suspend fun main() {
    val art: ASCIIArt = ai("ASCII art of a cat dancing")
    println(art.art)
}

File: examples/scala/src/main/scala/com/xebia/functional/xef/scala/conversation/serialization/ASCIIArt.scala
package com.xebia.functional.xef.scala.auto.serialization

import com.xebia.functional.xef.prompt.Prompt
import com.xebia.functional.xef.scala.auto.*
import io.circe.Decoder

private final case class ASCIIArt(art: String) derives SerialDescriptor, Decoder

@main def runASCIIArt: Unit =
  lazy val asciiArt = conversation {
    prompt[ASCIIArt](Prompt("ASCII art of a cat dancing"))
  }
  println(asciiArt.art)
